clmm/utils.py

```diff
@@ -159,35 +172,39 @@ def convert_units(dist1, unit1, unit2, redshift=None, cosmo=None):
         raise ValueError(f"Output units ({unit2}) not supported")
     # Try automated astropy unit conversion
     try:
-        return (dist1*units_bank[unit1]).to(units_bank[unit2]).value
+        dist2 = (dist1*units_bank[unit1]).to(units_bank[unit2]).value
     # Otherwise do manual conversion
     except u.UnitConversionError:
         # Make sure that we were passed a redshift and cosmology
         if redshift is None or cosmo is None:
-            raise TypeError("Redshift and cosmology must be specified to convert units")
+            raise TypeError(
+                "Redshift and cosmology must be specified to convert units") \
+                from u.UnitConversionError
         # Redshift must be greater than zero for this approx
         if not redshift > 0.0:
-            raise ValueError("Redshift must be greater than 0.")
+            raise ValueError("Redshift must be greater than 0.") from u.UnitConversionError
```

Comment:
What does the `from u.UnitConversionError` adds here, compared to what was there previously? Just out of curiosity as I haven't seen this used before.

Reply:
Forget about it, just tried it and saw the improved print out clearly indicating the type of error as `UnitConversionError` when using a bad redshift.